Core/include/Acts/Utilities/ContainerRange.hpp

```diff
@@ -74,6 +76,37 @@ class ContainerRange {
     return container().begin() + m_range.second;
   }
 
+  constexpr auto operator[](Index index) const noexcept {
```

Comment:
I think this could be a `requires` constraint, so that the range just does not have the relevant method.